open-apparel-registry/open-apparel-registry
# Add Settings Page

## Overview

Allows the user to access their profile, tokens, and embedded map
configuration under tabs in a new settings section. Adds the initial
embedded map configuration controls and preview.

Follow-up Tasks:
- Create API to manage Contributor Custom Fields. 
- Integrate API for Contributor Custom Fields to Embed Config Page. 
- Allow Contributor Custom Fields in Embedded Map.
- Make Contributor Custom Fields Draggable
- Allow Custom Color Theming
- Allow Custom Font
- Make Iframe Preview and Embed Code Responsive

Connects #1287 

## Demo

<img width="1083" alt="Screen Shot 2021-04-01 at 12 33 14 PM" src="https://user-images.githubusercontent.com/21046714/113326708-08a48600-92e8-11eb-805d-6d72d3003eb7.png">
<img width="511" alt="Screen Shot 2021-04-01 at 12 33 27 PM" src="https://user-images.githubusercontent.com/21046714/113326713-0a6e4980-92e8-11eb-8672-d6fe789eb39a.png">
<img width="911" alt="Screen Shot 2021-04-01 at 12 33 34 PM" src="https://user-images.githubusercontent.com/21046714/113326720-0b9f7680-92e8-11eb-93fc-888c6bd0e25e.png">

## Testing Instructions

- Navigate to a [contributor page](http://localhost:6543/profile/9) and confirm that you can see the profile as before
- Login and click the dropdown in the top-right corner. You should see ‘settings’ as an option. 
- Go to ’settings’. You should see your profile page under a ‘profile’ tab. You should see an embed tab and a tokens tab. 
- Select the tokens tab. You should be able to create and delete your token as before. 
- Select the embed tab. The embed controls should match the wireframe. 
- Confirm that the select all and individual checkboxes for fields behave as expected. 
- Change the text for a field. A ‘reset’ button should appear. Click the reset button and the filename should be restored to the default value. 
- Change the theme color. Confirm it is represented in the iframe code. 
- Select a font. Confirm that the fonts are previewed in the select. The selected font should be reflected in the iframe code. 
- Change the width and height of the iframe. This should be reflected in the embed code and the preview. 
- Select the ‘100%’ option. The width input should be disabled and the height input should have helper text and change to %. 
- You should see an iframe code with a copy button. Click the copy button and confirm the code was added to your clipboard.

## Checklist
